### ISSUE TITLE
Form submit does not work when a custom name is set for form

### ISSUE THREAD

**Issue by PhilippGrashoff** [atk4/ui#495]
Hi there,
when doing something like this:
```
$form->name = 'eoo-mainForm';
```

the callback is not found any more, the usual error message is displayed:
`!! Callback requested, but never reached. You may be missing some arguments in /TourManagement/places_new.php?eoo-mainForm_submit=ajax&__atk_callback=1`

**Comment by romaninsh** [comment]
@ibelar please look into this.

**Comment by PhilippGrashoff** [comment]
For what I want to do the easy solution was to set the custom ID to a container element. I wanted to have a custom ID to address it in JS. Using the container element works well.

**Comment by romaninsh** [comment]
ok, i think there is some minor inconsistency somewhere.

**Comment by PhilippGrashoff** [comment]
Yeah, name and id properties do some funny things when changing them. issue #456 is related. only changing the ID works in the html source code, but the js actions assigned to this view do not use the custom ID, this only works when name property is changed.

**Comment by ibelar** [comment]
This is a goog one!

The form create an hidden input named after the form name like this: formName_submit with value 'submit' and callback rely on that post request to properly call the callback function. 
In your case: 
```
<input id="eoo-mainForm_view" style="display:none" name="eoo-mainForm_submit" value="submit">
````
By default jquery-serialize-object does not allow '-' char to be use as input name. So it will simply remove this input from the post request. Since this hidden input is not send to the server, and callback rely on that input name and value, then callback is never reached.

There are two options:

1- simply remove the '-' char in your $form-name. For example this would work: 'eoo_mainForm'

2- if you really need to use '-' char then you would need to instruct jQuery serialize object plugin to use '-' char in your input name by adding this to your js file.

````
$.extend(FormSerializer.patterns, {
  validate: /^[a-z][a-z0-9_-]*(?:\[(?:\d*|[a-z0-9_-]+)\])*$/i,
  key:      /[a-z0-9_-]+|(?=\[\])/gi,
  named:    /^[a-z0-9_-]+$/i
});
````

**Comment by PhilippGrashoff** [comment]
Glad you enjoyed this one - I can simply change the ID and found a workaround already anyway.

**Comment by romaninsh** [comment]
i believe we should get rid of the '-' inside PHP code if it screws with jQuery. I remember a similar bug in the past with serializer.

**Comment by PhilippGrashoff** [comment]
I think adding some comment to View->id property to not use '-' sign should do the job, then this issue can be closed.

**Comment by ibelar** [comment]
This was fix with PR #530  
we can probably close this one.
